apps/provider-proxy/src/container.ts

```diff
@@ -2,33 +2,42 @@ import { LoggerService } from "@akashnetwork/logging";
 import { netConfig, SupportedChainNetworks } from "@akashnetwork/net";
 
 import { CertificateValidator, createCertificateValidatorInstrumentation } from "./services/CertificateValidator";
+import { HttpLoggerInterceptor } from "./services/HttpLoggerInterceptor";
 import { ProviderProxy } from "./services/ProviderProxy";
 import { ProviderService } from "./services/ProviderService";
 import { WebsocketStats } from "./services/WebsocketStats";
 
-const wsStats = new WebsocketStats();
-const providerService = new ProviderService((network: SupportedChainNetworks) => {
-  // TEST_CHAIN_NETWORK_URL is hack for functional tests
-  // there is no good way to mock external server in nodejs
-  // both nock and msw do not work well when I need to use low level API like X509 certificate validation
-  // for some reason when those libraries are used I receive MockSocket instead of TLSSocket
-  // @see https://github.com/mswjs/msw/discussions/2416
-  return process.env.TEST_CHAIN_NETWORK_URL || netConfig.getBaseAPIUrl(network);
-}, fetch);
-const certificateValidator = new CertificateValidator(
-  Date.now,
-  providerService,
-  process.env.NODE_ENV === "test" ? undefined : createCertificateValidatorInstrumentation(new LoggerService({ name: "cert-validator" }))
-);
-const providerProxy = new ProviderProxy(certificateValidator);
-const httpLogger = new LoggerService({ name: "http-proxy" });
-const createWsLogger = process.env.NODE_ENV === "test" ? undefined : (context: string) => new LoggerService({ name: "ws-proxy" }).setContext(context);
+export function createContainer() {
```

Comment:
this getting more and more into it. Perhaps using a container like in api would be a cleaner approach. non-blocking

Reply:
just to ensure that open-telementry can do its job before I instantiate some service. just function wrapper.

Reply:
decorator can be applied in the api or provided with a custom provider. there's similar approach with https://github.com/akash-network/console/blob/main/apps/api/src/core/providers/http-sdk.provider.ts